From d34d635a6e8723af2522ce64ee108c0eb5bde1c9 Mon Sep 17 00:00:00 2001
From: "penikelapati.kumar" <penikelapati.kumar@lge.com>
Date: Thu, 22 Jul 2021 16:26:18 +0530
Subject: [PATCH] Fix connman crash

:Release Notes:
Fix crash issue in connman

:Detailed Notes:
when p2p listen state diabled memory not freed properly so its
crashing in interface_removed.

:Testing Performed:
Build successfully.
Ethrenet/wifi connection successfully.

:QA Notes:
N/A

:Issues Addressed:
[WRN-3452] [RDX_CRASH][webos] /usr/sbin/connmand in n/a
           (libc.so.6 + 0x1ac26)

Upstream-Status: Inappropriate [webOS specific]
---
 plugins/wifi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/wifi.c b/plugins/wifi.c
index 35bd8ffd..1fc6158a 100644
--- a/plugins/wifi.c
+++ b/plugins/wifi.c
@@ -4219,7 +4219,7 @@ static void interface_removed(GSupplicantInterface *interface)
 		if (connman_technology_get_p2p_listen(p2p_technology) == false &&
 				!__connman_peer_get_connected_exists()) {
 			if (!connman_technology_get_enable_p2p_listen(p2p_technology))
-				return;
+				break;
 			err = apply_p2p_listen_on_iface(wifi, &params);
 			if (err == 0)
 				connman_technology_set_p2p_listen(p2p_technology, true);
